[Approaches]: Raindrops (#339)
exercism/powershell: exercises/practice/raindrops/.approaches/ordered-hashtable/content.md

# Using `ordered-hashtable` and `foreach`

```powershell
function Raindrops() {
    [CmdletBinding()]
    Param(
        [int]$Number
    )
    $table = [ordered]@{
        3 = "Pling"
        5 = "Plang"
        7 = "Plong"
    }
    $sounds = foreach ($key in $table.Keys) {
        if ($Number % $key -eq 0) {
            $table.$key
        }
    }
    $sounds.Count -eq 0 ? "$Number" : -join $sounds
}
```

> This approach loop through an ordered `hashtable`'s keys.

Reviewer: ```suggestion
This approach loops through an `ordered hashtable`'s keys.
```

> If the input number is divisible against a key, then use that key to look up the specific string in the `hashtable`.

Reviewer: ```suggestion
If the input number is divisible by a key, retrieve the string associated with the key from the hashtable.
```

> The loop can be assign into a variable to represent the final collection.

Reviewer: ```suggestion
The loop can be assigned to a variable to represent the final collection of sounds.
```

> If the collect is empty at the end, return the original number as string.

Reviewer: ```suggestion
If the collection is empty at the end, return the original number as string.
```
